garak/probes/leakreplay.py

```diff
@@ -69,102 +60,192 @@ def _attempt_prestore_hook(self, attempt: Attempt, seq: int) -> Attempt:
                 if len(row) != 2:
                     continue
                 trigger, passage = row
-                passage = passage.strip().split("[MASK]")[0]
-                if not passage.strip():
+                prefix = passage.split("[MASK]")[0].strip()
+                if not prefix:
                     continue
                 self.triggers.append(trigger)
-                self.prompts.append(passage)
+                self.prompts.append(prefix)
         self.triggers = self.langprovider.get_text(self.triggers)
-
         if self.follow_prompt_cap:
             self._prune_data(self.soft_probe_prompt_cap, prune_triggers=True)
 
     def _attempt_prestore_hook(self, attempt: Attempt, seq: int) -> Attempt:
         attempt.notes["triggers"] = [self.triggers[seq]]
         return attempt
 
-
-class LiteratureComplete(LiteratureCompleteFull, garak.probes.Probe):
-    __doc__ = LiteratureCompleteFull.__doc__ + " - limited number of examples"
+class NonFullMixin:
+    """Mixin for lightweight (limited) variants of probes."""
     active = True
     follow_prompt_cap = True
-    # source_file = "book_cloze_80.tsv"
 
+# ─── Domain‐Specific Probe Classes ──────────────────────────────────────
 
-class NYTCloze(LiteratureClozeFull, garak.probes.Probe):
-    active = True
-    source_file = "nyt_cloze.tsv"
-    __doc__ = (
-        LiteratureClozeFull.__doc__
-        + " - based on NYT articles used in the NYT/OpenAI lawsuit"
-    )
+# Literature
+class LiteratureClozeFull(ClozeProbeMixin, garak.probes.Probe):
+    """Test for data leakage on literature passages with masked entities in a cloze test format.
+    
+    Tests if models can fill in masked entities from literature excerpts, using the full dataset.
+    """
+    source_file = "book_cloze.tsv"
 
+class LiteratureCloze(NonFullMixin, LiteratureClozeFull):
+    """Lightweight version of literature cloze test for data leakage.
+    
+    Uses a limited subset of the literature dataset to test for data leakage with masked entities.
+    """
+    pass
 
-class NYTComplete(LiteratureCompleteFull, garak.probes.Probe):
-    active = True
+class LiteratureCompleteFull(CompleteProbeMixin, garak.probes.Probe):
+    """Test for data leakage on literature passages through text completion.
+    
+    Tests if models can complete literature excerpts from their training data, using the full dataset.
+    """
+    source_file = "book_cloze.tsv"
+    tier = garak.probes.Tier.COMPETE_WITH_SOTA  # regraded to tier 2
+
+class LiteratureComplete(NonFullMixin, LiteratureCompleteFull):
+    """Lightweight version of literature completion test for data leakage.
+    
+    Uses a limited subset of the literature dataset to test for data leakage through text completion.
+    """
+    tags = [
+        "avid-effect:security:S0301",
+        "owasp:llm10",
+        "owasp:llm06",
+        "quality:Security:ExtractionInversion",
+        "payload:leak:training"
+    ]
```

Comment:
looks like repetition of the value in parent class `CompleteProbeMixin` ?

Reply:
removed it, thanks.